### ISSUE TITLE
bottom control and Safari/iOS

### ISSUE THREAD

**Issue by cigone-openindoor** [maplibre/maplibre-gl-js#1253]
The bottom navigation bar in Safari/iOS cover bottom controls so they are unreachable, as shown in the above screenshot.

![signal-2022-05-28-060038](https://user-images.githubusercontent.com/69067574/170809153-1cd4f3a9-a7ed-4bcd-951a-4b49e1996575.png)

It would be nice to suggest a solution working also for Safari/iOS

**maplibre-gl-js version**: 2.1.9 (and any previous version)

**browser**: safari/iOS

**Comment by cigone-openindoor** [comment]
Finally, I fixed it by setting html CSS like this:
```
html {
    (...)
    height: 100vh;
    height: -moz-available;
    height: -webkit-fill-available;
    height: fill-available;
    (...)
}
```

**Comment by wipfli** [comment]
Nice, thanks @cigone-openindoor Should we close this issue or would you like to update the docs somewhere or something like that?

**Comment by cigone-openindoor** [comment]
OK, let’s close it.
However, it would be useful for any other developer who aim to have convenient support on web Apple world.
I don’t know where to put these info on the doc…

**Comment by wipfli** [comment]
Hm yes I also don't know a good place. This is not part of the MapLibre GL JS code base, right?

**Comment by wipfli** [comment]
Maybe make a variant of https://maplibre.org/maplibre-gl-js-docs/example/navigation/ with a bottom control?

**Comment by cigone-openindoor** [comment]
Maybe a new “responsive” section in the doc…

**Comment by wipfli** [comment]
By the way does this also affect chrome on android?

**Comment by tuukka** [comment]
Is there or could there be a project starter template that gets all the details like this right?

**Comment by wipfli** [comment]
That is a good idea @tuukka. Maybe we have already some in the awesome README at https://github.com/maplibre/awesome-maplibre

**Comment by tuukka** [comment]
> By the way does this also affect chrome on android?

Yes. `100vh` would be the most handy way to express full height, but mobile browsers don't subtract the height taken by the address bar.

The map at maplibre.org could work as an example, although it's not possible to see the issue as there is no bottom control: https://maplibre.org/

Also, since the site's top bar is transparent, it has been possible to get the right map height via `position: absolute; top: 0; bottom: 0;` https://github.com/maplibre/maplibre.github.io/blob/main/content/_index.html#L8

LINZ Basemaps (via Awesome MapLibre) has a top bar and a bottom control, but it uses flex to fill the available space in the layout: `flex: 1` 

In my own app I currently divide the space between the top bar and the map with `height: 10%` and `height: 90%` which is not the most convenient either but it works when the containing element is also full height.

In summary: there's no good generic example. Embracing flex is an option, and so is `fill-available` (like @cigone-openindoor wrote).

@cigone-openindoor Which documentation did you have a look at before and after deciding to use `100vh`? I guess one good place might be in README before the link to https://maplibre.org/maplibre-gl-js-docs/example/

**Comment by cigone-openindoor** [comment]
I did not check deeply all doc for all browser and O.S.
I just had a try on iOS (chrome, safari, firefo) and macOS (same browser).
